cli/command/service/scale.go

```diff
@@ -59,6 +70,18 @@ func runScale(dockerCli *command.DockerCli, args []string) error {
 		if err := runServiceScale(dockerCli, serviceID, scale); err != nil {
 			errs = append(errs, fmt.Sprintf("%s: %v", serviceID, err))
 		}
+
+		if options.detach {
+			if !flags.Changed("detach") && versions.GreaterThanOrEqualTo(dockerCli.Client().ClientVersion(), "1.29") {
+				fmt.Fprintln(dockerCli.Err(), "Since --detach=false was not specified, tasks will be scaled in the background.\n"+
```

Comment:
Instead of defining the error in a global we should make this a function and pass in the different string as an arg.

```go
func warnDetachDefault(dockerCli command.Cli, flags pflag.FlatSet, msg string) { ... }
```

We can use that function from all 3 locations, and it would be easy to test.